klock/src/commonMain/kotlin/com/soywiz/klock/PatternDateFormat.kt

```diff
@@ -263,15 +263,23 @@ data class PatternDateFormat @JvmOverloads constructor(
                 }
                 "MMMM" -> month = realLocale.months.indexOf(value) + 1
                 "MMMMM" -> if (doThrow) throw RuntimeException("Not possible to get the month from one letter.") else return null
-                "a" -> isPm = value == "pm"
+                "a" -> isPm = value.lowercase() == "pm"
```

Comment:
Can you use equals ignoreCase = true? https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/equals.html#equals
That will prevent an String instance creation when converting to lowercase.